dvc/objects/fs/path.py

```diff
@@ -80,7 +80,18 @@ def parents(self, path):
         )
 
     def name(self, path):
-        return self.parts(path)[-1]
+        if not path:
+            return ""
+
+        parent, name = self.flavour.split(path)
+
+        if name:
+            return name
+
+        if parent == path:
+            return ""
+
+        return self.name(parent)
```

Comment:
why not
```suggestion
        return self.flavour.basename(path)
```
then?

Reply:
But it breaks the tests that are testing arguably incorrect behaviour 😄 I just pointed out that old implementation was handling it differently. Matching `os.path.basename` behaviour is the correct thing to do, so we can just use it directly and not even have to test it specifically.